### PR TITLE
feat: Integrate Brave Web Search API in search toolkit (#1005)

### PR DESCRIPTION
**Pull request by MuggleJinx**
## Description

[Brave Web Search API](https://api.search.brave.com/app/documentation/web-search/get-started) is a REST API to query Brave Search and get back search results from the web. 
This PR enhance the search toolkit with the Brave Web Search API.

## Motivation and Context

Closes issue #1005.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [x] Add one new model type in Gemini model series.
- [x] Edit the corresponding example, and test file. 

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by liuxukun2000** [comment]
Hi @MuggleJinx! Thank you for your contribution! Could you please add a comment at the end of the example file to indicate the expected output for this example? You can refer to the other examples for guidance. Thanks!

File: camel/toolkits/search_toolkit.py [hunk 1]
```diff
@@ -151,6 +151,142 @@ def search_duckduckgo(
...
+        See https://api.search.brave.com/app/documentation/web-search/query
+        for more details.
+
+        Args:
+            q (str): The user's search query term. Query cannot be empty.
+                Maximum of 400 characters and 50 words in the query.
+            country (str): The search query country where results come from.
+                The country string is limited to 2 character country codes of
+                supported countries. For a list of supported values, see
+                Country Codes.
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:184]: add default value like below, same for other args
```suggestion
                Country Codes. (default::obj:`US `)
```

File: examples/toolkits/search_toolkit.py [hunk 2]
```diff
@@ -55,3 +56,25 @@
...
+        engine to answer questions.""",
+    tools=[FunctionTool(SearchToolkit().search_brave)],
+)
+
+usr_msg = "What is the temperature in Tokyo?"
+
+response = agent.step(input_message=usr_msg, response_format=None)
+
+print(response.msgs[0].content)
+print(response.info["tool_calls"])
```
Comment (Wendong-Fan) [inline examples/toolkits/search_toolkit.py:80]: add the printed output, you can refer to our example code for model part

File: camel/toolkits/search_toolkit.py [hunk 3]
```diff
@@ -151,6 +151,142 @@ def search_duckduckgo(
...
+                - 'metric': The standardized measurement system
+                - 'imperial': The British Imperial system of units.
+            extra_snippets (bool): A snippet is an excerpt from a page you get
+                as a result of the query, and extra_snippets allow you to get
+                up to 5 additional, alternative excerpts. Only available under
+                Free AI, Base AI, Pro AI, Base Data, Pro Data and Custom plans.
+            summary (bool): This parameter enables summary key generation in
+                web search results. This is required for summarizer to be
+                enabled.
+        """
```
Comment (Wendong-Fan) [inline camel/toolkits/search_toolkit.py:250]: missing docstring for `Returns`

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @MuggleJinx ! Overall LGTM, some minor change required

**Review by Wendong-Fan — APPROVED** [review]

**Pull request merged**